LoopKit/TemporaryScheduleOverride.swift

```diff
@@ -429,7 +429,88 @@ extension TemporaryScheduleOverride.EnactTrigger: Codable {
 
 extension Array where Element == TemporaryScheduleOverride {
 
-    public func apply<T>(
+    public func applySensitivity(over timeline: [AbsoluteScheduleValue<Double>]) -> [AbsoluteScheduleValue<Double>] {
+        apply(over: timeline) { value, override in
+            value / override.settings.effectiveInsulinNeedsScaleFactor
+        }
+    }
+
+    public func applySensitivity(over timeline: [AbsoluteScheduleValue<HKQuantity>]) -> [AbsoluteScheduleValue<HKQuantity>] {
+        apply(over: timeline) { quantity, override in
+            let value = quantity.doubleValue(for: .milligramsPerDeciliter)
+            return HKQuantity(
+                unit: .milligramsPerDeciliter,
+                doubleValue: value / override.settings.effectiveInsulinNeedsScaleFactor
+            )
+        }
+    }
+
+    public func applyBasal(over timeline: [AbsoluteScheduleValue<Double>]) -> [AbsoluteScheduleValue<Double>] {
+        apply(over: timeline) { value, override in
+            value * override.settings.effectiveInsulinNeedsScaleFactor
+        }
+    }
+
+    public func applyCarbRatio(over timeline: [AbsoluteScheduleValue<Double>]) -> [AbsoluteScheduleValue<Double>] {
+        apply(over: timeline) { value, override in
+            value * override.settings.effectiveInsulinNeedsScaleFactor
+        }
+    }
+
+    public func applyTarget(over timeline: [AbsoluteScheduleValue<ClosedRange<HKQuantity>>], at date: Date) -> [AbsoluteScheduleValue<ClosedRange<HKQuantity>>] {
+
+        guard timeline.count > 0 else {
+            return []
+        }
+
+        var applicableOverride: TemporaryScheduleOverride? = nil
+        let scheduleEndDate = timeline.last!.endDate
+
+        // Look for active or future override
+        for override in self {
+            if override.scheduledEndDate > date && override.startDate < scheduleEndDate {
+                // override is active or future
+                applicableOverride = override
```

Comment:
Should the array be searched from the end and then just break the loop after finding the desired override?

Reply:
Yes, I think a break makes sense; we should use the first applicable override. There should generally only be one future override, but still.